ftests/nuxeo-platform-explorer-ftests/explorer-ftests-base/src/main/java/org/nuxeo/functionaltests/explorer/testing/AbstractExplorerDownloadTest.java

```diff
@@ -0,0 +1,136 @@
+import org.nuxeo.runtime.api.Framework;
+import org.openqa.selenium.Proxy;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxProfile;
+import org.openqa.selenium.remote.CapabilityType;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import net.jsourcerer.webdriver.jserrorcollector.JavaScriptError;
+
+/**
+ * Abstract explorer test with download FF features and associated helpers.
+ *
+ * @since 11.1
+ */
+public abstract class AbstractExplorerDownloadTest extends AbstractExplorerTest {
+
+    public static File downloadDir;
+
+    public static String SAMPLE_BUNDLE_GROUP = "apidoc";
+
+    /**
+     * Updates the firefox profile to ease up testing of downloaded distribution.
+     * <p>
+     * See NXP-20646 for download solution.
+     */
+    @BeforeClass
+    public static void initFirefoxDriver() throws Exception {
+        assumeTrue(driver instanceof FirefoxDriver);
+
+        // quit existing: will recreate one
+        quitDriver();
+
+        proxyManager = new ProxyManager();
+        Proxy proxy = proxyManager.startProxy();
+        if (proxy != null) {
+            proxy.setNoProxy("");
+        }
+        DesiredCapabilities dc = DesiredCapabilities.firefox();
+        dc.setCapability(CapabilityType.PROXY, proxy);
+        FirefoxProfile profile = FirefoxDriverProvider.getProfile();
+        JavaScriptError.addExtension(profile);
+
+        // specific profile part
+        downloadDir = Framework.createTempDirectory("webdriver-explorer-admin").toFile();
+        profile.setPreference("browser.download.dir", downloadDir.toString());
+        profile.setPreference("browser.download.folderList", 2);
+        profile.setPreference("browser.download.useDownloadDir", true);
+        profile.setPreference("browser.helperApps.neverAsk.saveToDisk", "application/zip");
+
+        dc.setCapability(FirefoxDriver.PROFILE, profile);
+        driver = new FirefoxDriver(dc);
+    }
+
+    @AfterClass
+    public static void cleanupDownloadDir() throws Exception {
+        FileUtils.deleteDirectory(downloadDir);
+    }
+
+    protected static File createSampleZip(boolean addMarker) throws IOException {
+        String sourceDirPath = "data/sample_export";
+        File zip = new File(downloadDir, "distrib-apidoc.zip");
+        FileUtils.deleteQuietly(zip);
+        Path p = Files.createFile(Paths.get(zip.getPath()));
+        try (ZipOutputStream zs = new ZipOutputStream(Files.newOutputStream(p))) {
+            if (addMarker) {
+                ZipEntry zipEntry = new ZipEntry(".nuxeo-archive");
+                zs.putNextEntry(zipEntry);
+                zs.closeEntry();
+            }
+            // read paths from reference file as NuxeoArchiveReader requires a given order and extra info
+            Path epath = Paths.get(sourceDirPath, "entries.txt");
+            List<String> lines = IOUtils.readLines(AbstractExplorerTest.getReferenceStream(epath),
```

Comment:
I think `readLines` doesn't close the InputStream for us like the other methods could do, it could be interesting to check it.